pkgs/by-name/en/ente-web/package.nix

```diff
@@ -35,18 +35,36 @@ stdenv.mkDerivation (finalAttrs: rec {
     nodejs
   ];
 
+  env.NEXT_PUBLIC_ENTE_ENDPOINT = "NIXPKGS_BUILD_ENTE_PUBLIC_ENDPOINT";
+
   installPhase = ''
     cp -r apps/photos/out $out
   '';
 
-  meta = {
-    description = "Web client for Ente Photos";
-    homepage = "https://ente.io/";
-    license = lib.licenses.agpl3Only;
-    maintainers = with lib.maintainers; [
-      surfaceflinger
-      pinpox
-    ];
-    platforms = lib.platforms.all;
-  };
+  passthru.withEndpoint = { publicEndpoint }:
+    runCommand finalAttrs.finalPackage.name {
+      meta = {
+        description = "Web client for Ente Photos";
+        homepage = "https://ente.io/";
+        license = lib.licenses.agpl3Only;
+        maintainers = with lib.maintainers; [
+          surfaceflinger
+          pinpox
+        ];
+        platforms = lib.platforms.all;
+      };
+      env.NIXPKGS_BUILD_ENTE_PUBLIC_ENDPOINT = publicEndpoint;
+    } ''
+      cp -r --no-preserve=mode ${finalAttrs.finalPackage} $out
+      shopt -s globstar
+      readarray -t files < <(grep -rl NIXPKGS_BUILD_ENTE_PUBLIC_ENDPOINT $out)
+      if (( ''${#files[@]} == 0 )); then
+        echo "Failed to find endpoint replacement string"
+        exit 1
+      fi
+
+      substituteInPlace "''${files[@]}" \
+        --replace-fail NIXPKGS_BUILD_ENTE_PUBLIC_ENDPOINT "$NIXPKGS_BUILD_ENTE_PUBLIC_ENDPOINT"
+    '';
```

Comment:
```
nix-repl> ente-web.meta
{
  available = true;
  broken = false;
  insecure = false;
  name = "ente-web-0.9.5";
  outputsToInstall = [ ... ];
  position = "/home/hexa/git/nixos/master/pkgs/by-name/en/ente-web/package.nix:14";
  unfree = false;
  unsupported = false;
}
```

Not idea what the idea here is, but we're loosing proper metadata on the attribute.

Reply:
I changed it back to just using an override. If the user chooses to set an override, it might take around ~2 minutes to build without cache. I reconsidered the alternative and think the simpler, more maintainable package is worth the small burden on a user specifically setting this.